From: Colin Watson <cjwatson@debian.org>
Date: Tue, 29 Oct 2024 11:42:24 +0000
Subject: Fix import error on Python 3.13

Rework `escape_param` to work on Python 3.13.  One of `shlex.quote` or
`pipes.quote` exists on all Python versions (back to Python 0.9.8!), so
the fallback code for systems that have neither is unnecessary.

Forwarded: https://github.com/ansible-collections/community.mongodb/pull/684
Last-Update: 2024-10-29
---
 .../mongodb/plugins/module_utils/mongodb_shell.py   | 21 ++++++---------------
 1 file changed, 6 insertions(+), 15 deletions(-)

diff --git a/ansible_collections/community/mongodb/plugins/module_utils/mongodb_shell.py b/ansible_collections/community/mongodb/plugins/module_utils/mongodb_shell.py
index ec76686..eba0789 100644
--- a/ansible_collections/community/mongodb/plugins/module_utils/mongodb_shell.py
+++ b/ansible_collections/community/mongodb/plugins/module_utils/mongodb_shell.py
@@ -1,31 +1,22 @@
 from __future__ import absolute_import, division, print_function
 __metaclass__ = type
 
-# DeprecationWarning: 'pipes' is deprecated and slated for removal in Python 3.13
-# Ignore for now
-import warnings
-warnings.filterwarnings("ignore", category=DeprecationWarning)
-
-import shlex
-import pipes
 import re
 import json
 import os
 
+try:
+    from shlex import quote
+except ImportError:
+    from pipes import quote
+
 
 def escape_param(param):
     '''
     Escapes the given parameter
     @param - The parameter to escape
     '''
-    escaped = None
-    if hasattr(shlex, 'quote'):
-        escaped = shlex.quote(param)
-    elif hasattr(pipes, 'quote'):
-        escaped = pipes.quote(param)
-    else:
-        escaped = "'" + param.replace("'", "'\\''") + "'"
-    return escaped
+    return quote(param)
 
 
 def add_arg_to_cmd(cmd_list, param_name, param_value, is_bool=False, omit=None):
